### DISCUSSION TITLE
Shelly EM for single phase metering.

### DISCUSSION THREAD

**Discussion by tonythesticks** [fabian-lauer/dbus-shelly-3em-smartmeter#10] [Q&A]
Hi. I am soon installing a Multiplus II GX and found this code which looks ideal for what I want to do. My question is, will this code work for the single phase Shelly EM? I would like to use the two inputs to the device, one for grid and one for solar.

**Comment by Brekkis** [comment]
Here you'll find the Solution. 

https://community.victronenergy.com/questions/142851/shelly-em-single-phase-with-venus-os.html

Regards Stefan

**Comment by tonythesticks** [comment]
Perfect, thankyou!
________________________________

**Reply by Brekkis** [reply]
Please give a short Feedback if running correct...

**Comment by tonythesticks** [comment]
Will do.
________________________________

**Comment by superpoli** [comment]
I use this script in combination with a Shelly EM. "https://github.com/vincegod/dbus-shelly-em-smartmeter"
But I only see the existing inverter (integrated via Modbus) in the Victron main menu. The Shelly can only be found in the submenu, also with the current voltage and current values.
![01](https://user-images.githubusercontent.com/56473844/226346933-e92dcb7b-aa08-4653-99be-0e2a9c827bf8.jpg)
![02](https://user-images.githubusercontent.com/56473844/226347018-1401e9fb-7f5c-4fd5-9e94-79c4785bd868.jpg)
![03](https://user-images.githubusercontent.com/56473844/226347021-7e26bf5b-31c7-4ddb-bb88-68c3c437f313.jpg)

**Reply by gerdh** [reply]
you have to edit the py file and fill the total value in the last screen above ( I always take the same value as in the L1 line, since it is a one phase shelly anyway.).. see in the log file what is wrong. normally it is easy adaptable. around line 176...
